x-pack/plugins/infra/public/services/telemetry/telemetry_client.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { AnalyticsServiceSetup } from '@kbn/core-analytics-server';
+import { InfraTelemetryEventTypes, ITelemetryClient } from './types';
+
+/**
+ * Client which aggregate all the available telemetry tracking functions
+ * for the Infra plugin
+ */
+export class TelemetryClient implements ITelemetryClient {
+  constructor(private analytics: AnalyticsServiceSetup) {}
+
+  public reportHostEntryClicked: ITelemetryClient['reportHostEntryClicked'] = ({
```

Comment:
Would it be better if we just implement the interface? This way you don't need to refer back to `ITelemetryClient['reportHostEntryClicked']`

e.g

```

  public reportHostEntryClicked = ({
    hostname,
    cloud_provider: cloudProvider = 'unknown',
  }: HostEntryClickedParams) => {
    this.analytics.reportEvent(InfraTelemetryEventTypes.HOSTS_ENTRY_CLICKED, {
      hostname,
      cloud_provider: cloudProvider,
    });
  };
```